importers/data_sources/test/test_data_sources_importer.py

```diff
@@ -0,0 +1,205 @@
+
+
+
+class Test_data_sources_Importer(unittest.TestCase):
+    # @classmethod
+    # def setUp(cls):
+    #     # import the results file
+    #     # results_file = os.path.join(_TEST_DIR, "djornl", "results.json")
+    #     dir_path = os.path.dirname(os.path.realpath(__file__))
+    #     root_data_path = os.path.join(dir_path, "data")
+    #     with modified_environ(RES_ROOT_DATA_PATH=root_data_path):
+
+    def test_get_config(self):
+        imp = make_importer('standard')
+        self.assertIsNotNone(imp.get_config('API_URL', None))
+        self.assertIsNone(imp.get_config('foo', None))
+        imp = make_importer('standard', warm_config=False)
+        self.assertIsNotNone(imp.get_config('API_URL', None))
+
+    def test_get_config_or_fail(self):
+        imp = make_importer('standard')
+        self.assertIsNotNone(imp.get_config_or_fail('API_URL'))
+        with self.assertRaises(KeyError):
+            imp.get_config_or_fail('foo')
+        imp = make_importer('standard', warm_config=False)
+        self.assertIsNotNone(imp.get_config_or_fail('API_URL'))
+
+    def test_happy_load(self):
+        imp = make_importer('standard')
+        self.assertTrue(imp)
+        result = imp.load_data(dry_run=True)
+        self.assertTrue(result)
+
+    def test_happy_load_default_data_dir(self):
+        """Create importer which uses canonical data"""
+        imp = make_importer('standard', use_env_data=False)
+        result = imp.load_data(dry_run=True)
+        self.assertTrue(result)
+
+    def test_happy_load_bad_dir(self):
+        imp = make_importer('bad_dir')
+        result = imp.load_data(dry_run=True)
+        self.assertFalse(result)
+
+    def test_bad_ns(self):
+        imp = make_importer('bad_ns')
+        result = imp.load_data(dry_run=True)
+        self.assertFalse(result)
+        result = imp.load_data(dry_run=False)
+        self.assertFalse(result)
+
+    @responses.activate
+    def test_save_docs(self):
+        responses.add(responses.PUT, f'{API_URL}/api/v1/documents',
+                      json={'do_not': 'care'}, status=200)
+        imp = make_importer('standard')
+        result = imp.load_data(dry_run=False)
+        self.assertTrue(result)
+
+    @responses.activate
+    def test_save_docs_error(self):
+        responses.add(responses.PUT, f'{API_URL}/api/v1/documents',
+                      json={'do_not': 'care'}, status=400)
+        imp = make_importer('standard')
+        with self.assertRaises(RuntimeError):
+            result = imp.load_data(dry_run=False)
+
+    def test_do_import_dry_run(self):
+        do_import(dry_run=True)
+        self.assertTrue(True)
+
+    @responses.activate
+    def test_do_import_dry_run(self):
+        responses.add(responses.PUT, f'{API_URL}/api/v1/documents',
+                      json={'do_not': 'care'}, status=200)
+        do_import(dry_run=False)
+        self.assertTrue(True)
+
+    @responses.activate
+    def test_do_import_dry_run(self):
```

Comment:
I'm a bit surprised python doesn't care that this function has the same name as the two functions above!

Reply:
have been renamed.